### ISSUE TITLE
join_radar function is broken

### ISSUE THREAD

**Issue by jjhelmus** [ARM-DOE/pyart#127]
The function pyart.io.radar.join_radar is broken and does not work as intended.  The function also needs to be documented and added to the pyart.io namespace.

**Comment by jjhelmus** [comment]
This functionality should be in the pyart.util namespace, changing label to reflect this.

**Comment by nguy** [comment]
To revisit this topic. Do the [following lines](https://github.com/ARM-DOE/pyart/blob/master/pyart/util/radar_utils.py#L153-157) make sense? It in essence concatenates the azimuth dimension, but not the range dimension. I think this might fail for different location radars, only working for the same radar that uses the same scan strategy.

**Comment by jjhelmus** [comment]
I think those lines are reasonable but the subtraction of -9999 from an array of zeros does not guarentee masking of elements outside of the two radars.  Creating a fully masked array initially would likely be a better solution:

``` Python
fill_value = -9999.0  # should be retrieved from field metadata
sh1 = radar1.fields[var]['data'].shape
sh2 = radar2.fields[var]['data'].shape
new_field_shape = (sh1[0] + sh2[0], max(sh1[1], sh2[1]))
new_field = np.ma.masked_equal(np.ones(new_field_shape), fill_value)
new_field[0:sh1[0], 0:sh1[1]] = radar1.fields[var]['data']
new_field[sh1[0]:, 0:sh2[1]] = radar2.fields[var]['data']
new_radar.fields[var]['data'] = new_field
```

**Comment by nguy** [comment]
I agree that is more robust. 
How about [these lines](https://github.com/ARM-DOE/pyart/blob/master/pyart/util/radar_utils.py#L135-138) that do not concatenate the range. I don't follow the logic behind taking the greater length array?

**Comment by jjhelmus** [comment]
Those lines assume that the gate spacing and first gate of each radar are identical (see the comment at the beginning of the function).  Then the range data with the greatest length is the best solution for the combined radar.  If one of the radars has fewer gates than the other then gates beyond that point should be masked in the field data.  

CfRadial version 1.4 will introduce a specification for describing radar volumes where the range spacing and/or the number of range bins in each sweep changes.  Once this standard is out it should be easier to update the `join_radar` function to be more robust. Additionally, I think making the Radar class  hierarchical with added **Sweep** and **Ray** classes, #242, could make this process more elegant.

**Comment by nguy** [comment]
I see that note now, totally missed it. Yeah, I recall this conversation at the recent NetCDF meeting. 
I like the idea in #242 and will post there later after fully reading through. I've run into issues when processing multiple radar files and I need to think about how each method would affect future RadarCollection (yep, still thinking about this) operations.

Do you want to make the changes you mention above?

**Comment by jjhelmus** [comment]
I started a [branch](https://github.com/jjhelmus/pyart/tree/sweep_class) to implement some of the ideas in #242.  Still a work in progress and will likely be rebased before merging but if you are interested you can check it out.  

I would prefer to wait until CfRadial 1.4 comes out with the option two dimensional range variable before making updates to the `join_radar` function.  Once that option is available and Py-ART supports it the function will be more straightforward.

**Comment by jjhelmus** [comment]
Jordi Figueras i Ventura sent me some update for the `join_radar` function which I modified and included in PR #547.  This also implements some of the changes mentioned in this issue, although addition work will likely still be needed.

**Comment by zssherman** [comment]
Closing for now,
